src/internal/agent/proxy/transform.go

```diff
@@ -0,0 +1,115 @@
+package proxy
+
+import (
+	"fmt"
+	"hash/crc32"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/internal/agent/state"
+	"github.com/defenseunicorns/zarf/src/internal/message"
+	"github.com/defenseunicorns/zarf/src/types"
+)
+
+const (
+	// NoTransform is the URL prefix added to HTTP 3xx or text URLs that instructs Zarf not to transform on a subsequest request.
+	NoTransform = "/zarf-3xx-no-transform"
+)
+
+// GetProxyState gets the ZarfState and the NPM token for the proxy to use for a request.
+func GetProxyState() (types.ZarfState, string) {
+	zarfState, err := state.GetZarfStateFromAgentPod()
+	if err != nil {
+		message.Debugf("Unable to load the ZarfState file so that the Agent can mutate pods: %#v", err)
```

Comment:
This should probably return with an err or kill the application. 

Simply debugging will just end up with a nil pointer deference panic later anyways (either later in this func or by the func that calls this). Even if we just end up fatal'ing manually, it will be a lot prettier and more clear whats happening than getting a nil pointer stack trace.

Reply:
Generated and returned an error